cpp/src/io/parquet/reader_impl_chunking.hpp

```diff
@@ -47,6 +47,11 @@ struct file_intermediate_data {
   // partial sum of the number of rows per data source
   std::vector<std::size_t> exclusive_sum_num_rows_per_source{};
 
+  size_type num_input_row_groups{0};  // total number of input row groups across all data sources
+
+  // struct containing the number of remaining row groups after each predicate pushdown filter
+  num_surviving_row_groups remaining_row_groups;
```

Comment:
My objection to calling a struct `num_surviving_row_groups` originated here.

Reply:
I agree with you on this. Would `surviving_row_group_metrics` make more sense?

Reply:
I'd prefer it myself.  But this is stylistic; so I'm loath to hold you to it.

Reply:
It's okay, I already have the changes. Let me just push them while at it!